content/guides/dapps/cash-app.md

```diff
@@ -0,0 +1,2505 @@
+to simplify a lot of the front end:
+
+- `getDomainKeySync` - a function that returns the public key associated with
+  the provided domain name. This can be implemented anywhere there is a user
+  input for a public key. Now the user only needs to type in a username when
+  searching for an account, exactly as you do with Cash App. This is what SNS
+  calls a
+  [direct lookup](https://sns.guide/domain-name/domain-direct-lookup.html).
+
+- `reverseLookup` - an asynchronous function that returns the domain name of the
+  provided public key.This can be implemented anywhere in the UI where you want
+  to display the username. This is what SNS calls a
+  [reverse lookup](https://sns.guide/domain-name/domain-reverse-lookup.html)
+
+To showcase this, lets update the transfers funds function to now accept a user
+name as a parameter rather than a public key and integrate the SNS API.
+
+```tsx
+const transferFunds = useCallback(
+  async (program: Program<CashApp>) => {
+    let signedTransactions = await transact(
+      async (wallet: Web3MobileWallet) => {
+        const [authorizationResult, latestBlockhash] = await Promise.all([
+          authorizeSession(wallet),
+          connection.getLatestBlockhash(),
+        ]);
+
+        const { pubkey } = getDomainKeySync(userName);
+
+        const [recipientPDA] = useMemo(() => {
+          const recipientSeed = pubkey.toBuffer();
+          return PublicKey.findProgramAddressSync(
+            [recipientSeed],
+            cashAppProgramId,
+          );
+        }, [cashAppProgramId]);
+
+        const transferInstruction = await program.methods
+          .transferFunds(pubkey, newTransferAmount)
+          .accounts({
+            user: authorizationResult.publicKey,
+            fromCashAccount: cashAppPDA,
+            toCashAccount: recipientPDA,
+          })
+          .instruction();
+
+        const transferTransaction = new Transaction({
+          ...latestBlockhash,
+          feePayer: authorizationResult.publicKey,
+        }).add(transferInstruction);
+
+        const signedTransactions = await wallet.signTransactions({
+          transactions: [transferTransaction],
+        });
+
+        return signedTransactions[0];
+      },
+    );
+
+    let txSignature = await connection.sendRawTransaction(
+      signedTransactions.serialize(),
+      {
+        skipPreflight: true,
+      },
+    );
+
+    const confirmationResult = await connection.confirmTransaction(
+      txSignature,
+      "confirmed",
+    );
+
+    if (confirmationResult.value.err) {
+      throw new Error(JSON.stringify(confirmationResult.value.err));
+    } else {
+      console.log("Transaction successfully submitted!");
+    }
+  },
+  [authorizeSession, connection, cashAppPDA],
+);
+```
+
+This implementation can be integrated everywhere in the application where an
+input requires a public key, enabling the user experience to be identical to
+that of a web2 application.
```

Comment:
Any closing thoughts? A congratulations to the reader for making it this far and building their first mobile app?